specs/Print.md

```diff
@@ -0,0 +1,816 @@
+              }
+            }
+
+            AsyncMessageBox(message, L"Print to printer");
+
+            return S_OK;
+          })
+          .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToPrinter()
+{
+  string printerName = GetPrinterName();
+  CoreWebView2PrintSettings printSettings = GetSelectedPrinterPrintSettings(printerName);
+  try
+  {
+    string title = webView.CoreWebView2.DocumentTitle;
+
+    bool isSuccessful = await webView.CoreWebView2.PrintAsync(printSettings);
+
+    if (isSuccessful)
+    {
+       MessageBox.Show(this, "Printing " + title + " document to printer is succeeded", "Print to printer");
+    }
+    else
+    {
+      MessageBox.Show(this, "Selected printer is not found, not available, offline or error state.", "Print to printer");
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Printing is failed: " + exception.Message, "Print to printer");
+  }
+}
+
+// Function to get printer name by displaying printer text input dialog to the user.
+// User has to specify the desired printer name by querying the installed printers list on the
+// OS to print the web page.
+// You may also choose to display printers list to the user and return user selected printer.
+string GetPrinterName()
+{
+  var dialog = new TextInputDialog(
+                title: "Printer Name",
+                description: "Specify a printer name from the installed printers list on the OS.",
+                defaultInput: "");
+  if (dialog.ShowDialog() == true)
+  {
+    printerName = dialog.Input.Text;
+  }
+  return printerName;
+
+  // or
+  //
+  // Use GetPrintQueues() of LocalPrintServer from System.Printing to get list of locally installed printers.
+  // Display the printer list to the user and get the desired printer to print.
+  // Return the user selected printer name.
+}
+
+// Function to get print settings for the selected printer.
+// You may also choose get the capabilties from the native printer API, display to the user to get
+ // the print settings for the current web page and for the selected printer.
+CoreWebView2PrintSettings GetSelectedPrinterPrintSettings(string printerName)
+{
+  CoreWebView2PrintSettings printSettings = null;
+  printSettings = WebViewEnvironment.CreatePrintSettings();
+  printSettings.ShouldPrintBackgrounds = true;
+  printSettings.ShouldPrintHeaderAndFooter = true;
+
+  return printSettings;
+
+  // or
+  // Get PrintQueue for the selected printer and use GetPrintCapabilities() of PrintQueue from System.Printing
+  // to get the capabilities of the selected printer.
+  // Display the printer capabilities to the user along with the page settings.
+  // Return the user selected settings.
+}
+```
+
+## PrintToPdfStream
+
+You can use `PrintToPdfStream` method to display as a preview in a custom print dialog.
+The app can also use this API to configure the custom print preview dialog unlike confining
+print preview dialog to the webview2 control.
```

Comment:
Not sure what this sentence is trying to say.

Reply:
Some of our customers reported that print dialog is only displayed in the webview control (Print Preview window stays in the constraints of the WebView2 control). If the control is small, they cannot see the print area properly. 

To solve this problem, host app can build a custom print preview dialog using these APIs. 

This statement probably should be in background section. Will update.

Reply:
Should there be another API that lets the WV2 put a print preview window into a window that the app controls? Or maybe that's this API, but it's a pain, because the app now has to have a small WV2 with the content and a larger WV2 with the preview window?

Reply:
We have a third option that they can create their own print preview window or use the system print dialog. We're not planning on adding a fourth option of making the 'small webview' issue better.